tests/ui/needless_borrows_for_generic_args.stderr

```diff
@@ -37,41 +37,29 @@ error: the borrowed expression implements the required traits
 LL |         multiple_constraints_normalizes_to_same(&X, X);
    |                                                 ^^ help: change this to: `X`
 
-error: the borrowed expression implements the required traits
-  --> tests/ui/needless_borrows_for_generic_args.rs:127:24
-   |
-LL |             takes_iter(&mut x)
-   |                        ^^^^^^ help: change this to: `x`
-
```

Comment:
Is this also an expected change?

Reply:
Hmm... I see, I got confused between `warn` and `dont_warn` functions, it's probably better to keep this warning. Will try to fix.

Reply:
Added a check for that - taking a reference to a reference isn't helpful, rebased onto master just in case.